src/rgw/rgw_sal_motr.cc

```diff
@@ -452,8 +452,67 @@ int MotrBucket::remove_bucket(const DoutPrefixProvider *dpp, bool delete_childre
 {
   int ret;
 
+  // Algorithm: (Create Bucket Reversal)
+  // 1. unlink user.
+  // 2. Abort Mp uploads and remove mp index.
+  // 3. Remove Lifecycle config and sync user stats.
+  // 4. Remove bucket index.
+  // 5. Remove bucket instance info.
+  // 6. Remove Notifications attached to bucket.
+  // 7. Forward to master.
+
+  ldpp_dout(dpp, 20) << "remove_bucket Entry=" << info.bucket.name << dendl;
+
+  // Refresh info
   ret = load_bucket(dpp, y);
-  // TODO: implement bucket removal
+  if (ret < 0) {
+    ldpp_dout(dpp, 0) << "ERROR: remove_bucket load_bucket failed rc=" << ret << dendl;
+    return ret;
+  }
+
+  ListParams params;
+  params.list_versions = true;
+  params.allow_unordered = true;
+
+  ListResults results;
+
+  results.objs.clear();
+
+  // TBD : support deletion of bucket with objects.
```

Comment:
This flag should be checked if we find bucket is not empty .. so should be shifted inside below if block
if (!results.objs.empty())